src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java

```diff
@@ -455,6 +503,26 @@ public boolean sessionHasQuestions(String feedbackSessionName, String courseId)
                 }
             }
             break;
+        case TEAMS_IN_SAME_SECTION:
+            Map<String, List<StudentAttributes>> teamToTeamInSectionMembersTable;
+            List<StudentAttributes> studentsInTeamForSection =
+                    studentsLogic.getStudentsForSection(giverSection, question.getCourseId());
+            teamToTeamInSectionMembersTable = CourseRoster.buildTeamToMembersTable(studentsInTeamForSection);
+            for (Map.Entry<String, List<StudentAttributes>> team : teamToTeamInSectionMembersTable.entrySet()) {
+                if (isInstructorGiver && !instructorGiver.isAllowedForPrivilege(
+                        team.getValue().iterator().next().getSection(),
+                        question.getFeedbackSessionName(),
+                        Const.InstructorPermissions.CAN_SUBMIT_SESSION_IN_SECTIONS)) {
+                    // instructor can only see teams in allowed sections for him/her
+                    continue;
+                }
+                // Ensure student('s team) does not evaluate own team.
+                if (!giverTeam.equals(team.getKey())) {
+                    // recipientEmail doubles as team name in this case.
+                    recipients.put(team.getKey(), team.getKey());
+                }
+            }
```

Comment:
This part is identical from the previous case, might have room for refactoring. Perhaps let the `TEAMS` case fall through to  the `TEAMS_IN_SAME_SECTION` case and check for `generateOptionsFor == FeedbackParticipantType.TEAMS_IN_SAME_SECTION` / `generateOptionsFor == FeedbackParticipantType.TEAMS`. Similarly for the `STUDENTS_IN_SAME_SECTION` case above.

Reply:
I have updated the code as you have suggested, thank you!